experiment/migrate_testgrid_tabs.py

```diff
@@ -142,55 +216,89 @@ def find_prow_job(name, path):
-                annotation["testgrid-dashboards"] += (", " + dashboardName)
+                annotation["testgrid-dashboards"] += (", " + dashboard_name)
         else:
             #Existing annotation is non-testgrid-related
-            annotation["testgrid-dashboards"] = dashboardName
+            annotation["testgrid-dashboards"] = dashboard_name
 
     else:
         # There is no annotation; construct it
-        annotation = {"testgrid-dashboards": dashboardName}
+        annotation = {"testgrid-dashboards": dashboard_name}
 
     # Append optional annotations
-    if ("name" in dashTab
+    if ("name" in dash_tab
             and "testgrid-tab-name" not in annotation
-            and dashTab["name"] != dashTab["test_group_name"]):
-        annotation["testgrid-tab-name"] = dashTab["name"]
+            and dash_tab["name"] != dash_tab["test_group_name"]):
+        annotation["testgrid-tab-name"] = dash_tab["name"]
 
-    if ("alert_options" in dashTab
-            and "alert_mail_to_addresses" in dashTab["alert_options"]
+    if ("alert_options" in dash_tab
+            and "alert_mail_to_addresses" in dash_tab["alert_options"]
             and "testgrid-alert-email" not in annotation):
-        annotation["testgrid-alert-email"] = dashTab["alert_options"]["alert_mail_to_addresses"]
+        annotation["testgrid-alert-email"] = dash_tab["alert_options"]["alert_mail_to_addresses"]
 
     opt_arguments = [("description", "description"),
                      ("num_failures_to_alert", "testgrid-num-failures-to-alert"),
                      ("alert_stale_results_hours", "testgrid-alert-stale-results-hours"),
                      ("num_columns_recent", "testgrid-num-columns-recent")]
 
-    for tabName, annotationName in opt_arguments:
-        if (tabName in dashTab and annotationName not in annotation):
+    for tab_arg, annotation_arg in opt_arguments:
+        if (tab_arg in dash_tab and annotation_arg not in annotation):
+            # Numeric arguments need to be coerced into strings to be parsed correctly
+            annotation[annotation_arg] = str(dash_tab[tab_arg])
+
+    prow_yaml["annotations"] = annotation
+    return prow_yaml
+
+
+def patch_prow_job_with_group(prow_yaml, test_group, force_annotation=False):
+    """Updates a prow YAML object
+
+    Assumes a valid prow yaml and a compatible test group
+    Will amend existing annotations or create one if there is data to migrate
+    If there is no migratable data, an annotation will be forced only if specified
+    """
+    if "annotations" in prow_yaml or force_annotation:
+        # There exists an annotation; amend it
+        annotation = prow_yaml["annotations"]
+    else:
+        annotation = {}
+
+    # migrate info
+    opt_arguments = [("num_failures_to_alert", "testgrid-num-failures-to-alert"),
+                     ("alert_stale_results_hours", "testgrid-alert-stale-results-hours"),
+                     ("num_columns_recent", "testgrid-num-columns-recent")]
```

Comment:
don't have to fix this, just noticing the copy-pasting of these may lead us into typo territory

it might be useful to have a dict of testgrid->prowjob names and/or some constants